src/java.base/share/classes/java/io/FilterOutputStream.java

```diff
@@ -200,4 +200,13 @@ public void close() throws IOException {
             }
         }
     }
+
+    /**
+     * Tells whether or not this stream is open.
+     *
+     * @return {@code true} if, and only if, this stream is open
+     */
+    boolean isOpen(){
+        return !closed;
```

Comment:
Wouldn't you want `synchronized (closeLock)` here?

Reply:
'closed' is volatile, we don't need synchronized (closeLock) here.